cabal-install/Distribution/Client/Get.hs

```diff
@@ -314,20 +314,23 @@ branchDarcs = Brancher "darcs" $ \repo -> do
 branchGit :: Brancher
 branchGit = Brancher "git" $ \repo -> do
     src <- PD.repoLocation repo
-    let branchArgs = case PD.repoBranch repo of
-         Just b -> ["--branch", b]
-         Nothing -> []
     let postClone dst = case PD.repoTag repo of
          Just t -> do
              cwd <- getCurrentDirectory
              setCurrentDirectory dst
              finally
-                 (rawSystem "git" (["checkout", t] ++ branchArgs))
+                 (rawSystem "git" (["checkout", t] ++
+                    case PD.repoBranch repo of
```

Comment:
IIUC, we don't need to specify the branch here, since at this point we should've already checked out the correct one.